lib/chat_web/live/main_live/page/lobby.ex

```diff
@@ -91,22 +89,40 @@ defmodule ChatWeb.MainLive.Page.Lobby do
       PubSub.broadcast!(
         Chat.PubSub,
         @topic,
-        :room_request
+        {:room_request, room_hash, me |> Identity.pub_key() |> Utils.hash()}
       )
     end)
 
     socket
     |> assign_room_list()
   end
 
-  def approve_requests(socket) do
+  def approve_room_request(%{assigns: %{room_map: room_map}} = socket, room_hash, user_hash) do
+    if room_map[room_hash] do
+      Rooms.approve_request(room_hash, user_hash, room_map[room_hash])
+
+      PubSub.broadcast!(Chat.PubSub, @topic, {:room_request_approved, room_hash, user_hash})
+    end
+
     socket
-    |> approve_joined_room_requests()
   end
 
-  def join_rooms(socket) do
-    socket
-    |> join_approved_rooms()
+  def join_approved_room(
+        %{assigns: %{rooms: rooms, me: me, my_id: my_id, monotonic_offset: time_offset}} = socket,
+        room_hash,
+        user_hash
+      ) do
+    if my_id == user_hash do
+      time = Chat.Time.monotonic_to_unix(time_offset)
+      joined_room = Rooms.join_approved_requests(room_hash, me, time)
```

Comment:
`Rooms.join_approved_request` takes encrypted  key from  DB. We should be able to get room key from broadcasted message as well